tools/validate.py

```diff
@@ -11,75 +15,173 @@ def load_validator_from_url(uri_env_name):
     This is then loaded into jsonschema validator and returned.
     """
     schema_uri = os.environ.get(uri_env_name)
-    schema = requests.get(schema_uri).json()
-
-    return jsonschema.Draft7Validator(schema)
 
+    if schema_uri is None:
+        raise ValueError("Schema is not defined in base layer.")
 
-request_validator = load_validator_from_url("REQUEST_SCHEMA_URL")
-response_validator = load_validator_from_url("RESPONSE_SCHEMA_URL")
-
-#with open('/Users/louismanestar/OneDrive - Imperial College London/Jobs/Lambda Feedback/repos/lambda-feedback/request-response-schemas/request.json', 'r') as f:
-#    request_schema = json.load(f)
-#    request_validator = jsonschema.Draft7Validator(request_schema)
-
-#with open('/Users/louismanestar/OneDrive - Imperial College London/Jobs/Lambda Feedback/repos/lambda-feedback/request-response-schemas/response.json', 'r') as f:
-#    response_schema = json.load(f)
-#    response_validator = jsonschema.Draft7Validator(response_schema)
+    schema = requests.get(schema_uri).json()
+    return jsonschema.Draft7Validator(schema)
 
 
 def validate(validator, body):
     try:
         validator.validate(body)
     except jsonschema.exceptions.ValidationError as e:
-
         return {
             "message": e.message,
             "schema_path": list(e.absolute_schema_path),
-            "instance_path": list(e.absolute_path)
+            "instance_path": list(e.absolute_path),
         }
 
     return None
 
 
 def validate_request(body):
-    """    
+    """
+    Function to return any errors in the request body based on its schema.
+    ---
+    If there are no issues with the request body, then None is returned. Otherwise, a
+    JSON-encodable response containing the schema and errors will be returned. Each
+    element in the list of errors is a dictionary containing the error message and the
+    path to the rule in the schema that has thrown the error.
+    """
+    validator = load_validator_from_url("REQUEST_SCHEMA_URL")
+    request_error = validate(validator, body)
+
+    if request_error:
+        return {
+            "message": "Schema threw an error when validating the request body.",
+            "error_thrown": request_error,
+        }
```

Comment:
Extract common validation error response in function:

```python

def request_validation_error(error):
    return {
            "message": "Schema threw an error when validating the request body.",
            "error_thrown": error,
        }
```

In this way, your validation functions can be simplified with 
`return request_validation_error(request_error) if request_error else None`


Same applies for a similar `response_validation_error`